asm_to_pil/src/lib.rs

```diff
@@ -122,43 +56,19 @@ struct ASMPILConverter<'a, T> {
     rom_constant_names: Vec<String>,
 }
 
-impl<'a, T: FieldElement> ASMPILConverter<'a, T> {
-    fn new(location: &'a Location, machines: &'a BTreeMap<String, Machine<T>>) -> Self {
-        Self {
-            location,
-            machines,
-            pil: Default::default(),
-            pc_name: Default::default(),
-            registers: Default::default(),
-            submachines: Default::default(),
-            instructions: Default::default(),
-            code_lines: Default::default(),
-            line_lookup: Default::default(),
-            rom_constant_names: Default::default(),
-        }
-    }
-
-    fn handle_inline_pil(&mut self, PilBlock { statements, .. }: PilBlock<T>) {
-        self.pil.extend(statements)
-    }
-
-    fn convert_machine(mut self, input: Machine<T>) -> Object<T> {
-        let degree = input.degree.map(|s| T::from(s.degree.clone()).to_degree());
-
-        self.submachines = input.submachines;
-
+impl<T: FieldElement> ASMPILConverter<T> {
+    fn convert_machine(mut self, mut input: Machine<T>) -> Machine<T> {
```

Comment:
The signature of this function changed. What does it convert a Machine into?

Reply:
It turns a virtual machine into a constrained machine. It used to go all the way to AIR, but now there are more processing steps before we go to air in airgen.